[MSVC] Failed to build GDAL on windows with msvc due to "error LNK2019: unresolved external symbol libiconv_open referenced in function main"

### What is the bug?

I built GDAL on Windows with MSVC, it failed to build due to below errors, can you take a look? Thanks.
```
character_set_conv_table_generator.obj : error LNK2019: unresolved external symbol libiconv_open referenced in function main [C:\gitP\OSGeo\gdal\build_amd64\port\character_set_conv_table_generator.vcxproj]
character_set_conv_table_generator.obj : error LNK2019: unresolved external symbol libiconv referenced in function main [C:\gitP\OSGeo\gdal\build_amd64\port\character_set_conv_table_generator.vcxproj]
character_set_conv_table_generator.obj : error LNK2019: unresolved external symbol libiconv_close referenced in function main [C:\gitP\OSGeo\gdal\build_amd64\port\character_set_conv_table_generator.vcxproj]
C:\gitP\OSGeo\gdal\build_amd64\port\Release\character_set_conv_table_generator.exe : fatal error LNK1120: 3 unresolved externals [C:\gitP\OSGeo\gdal\build_amd64\port\character_set_conv_table_generator.vcxproj]
character_set_conv_table_generator.obj : error LNK2019: unresolved external symbol libiconv_open referenced in function main [C:\gitP\OSGeo\gdal\build_amd64\port\character_set_conv_table_generator.vcxproj]
character_set_conv_table_generator.obj : error LNK2019: unresolved external symbol libiconv referenced in function main [C:\gitP\OSGeo\gdal\build_amd64\port\character_set_conv_table_generator.vcxproj]
character_set_conv_table_generator.obj : error LNK2019: unresolved external symbol libiconv_close referenced in function main [C:\gitP\OSGeo\gdal\build_amd64\port\character_set_conv_table_generator.vcxproj]
C:\gitP\OSGeo\gdal\build_amd64\port\Release\character_set_conv_table_generator.exe : fatal error LNK1120: 3 unresolved externals [C:\gitP\OSGeo\gdal\build_amd64\port\character_set_conv_table_generator.vcxproj]
```

### Steps to reproduce the issue

1. git clone https://github.com/OSGeo/gdal C:\gitP\OSGeo\gdal
2.  vcpkg install --recurse **proj libiconv** --triplet x64-windows
3. mkdir C:\gitP\OSGeo\gdal\build_amd64 and cd C:\gitP\OSGeo\gdal\build_amd64
4. cmake -G "Visual Studio 17 2022" -A x64 -DCMAKE_SYSTEM_VERSION=10.0.22621.0 -DGDAL_BUILD_OPTIONAL_DRIVERS=OFF -DOGR_BUILD_OPTIONAL_DRIVERS=OFF -DCMAKE_CXX_COMPILER_LAUNCHER="clcache" -DCMAKE_UNITY_BUILD=OFF -DBUILD_TESTING=OFF -DCMAKE_TOOLCHAIN_FILE=C:\gitP\Microsoft\vcpkg\scripts\buildsystems\vcpkg.cmake -DVCPKG_TARGET_TRIPLET=x64-windows ..
5. set include=C:\gitP\Microsoft\vcpkg/installed/x64-windows/include;%include%
6. set libpath=C:\gitP\Microsoft\vcpkg/installed/x64-windows/debug/lib;%libpath%
7. msbuild /m /p:Platform=x64 /p:Configuration=Release gdal.sln /t:Rebuild /p:useenv=true


### Versions and provenance

OS: Windows Server 2022 21H2
GDAL: the latest commit 9c27570

### Additional context

[Build.log](https://github.com/user-attachments/files/17304956/Build.log)
[Setup.log](https://github.com/user-attachments/files/17304957/Setup.log)

--- a/port/CMakeLists.txt
+++ b/port/CMakeLists.txt
@@ -375,11 +375,24 @@ if (NOT CMAKE_CROSSCOMPILING AND BUILD_VSIPRELOAD AND "${CMAKE_SYSTEM}" MATCHES 
   endif ()
 endif ()
 
-# Utility to generate cpl_character_sets.h and .c
-add_executable(character_set_conv_table_generator EXCLUDE_FROM_ALL character_set_conv_table_generator.c)
+if (GDAL_USE_ICONV)
+    # Utility to generate cpl_character_sets.h and .c
+    add_executable(character_set_conv_table_generator EXCLUDE_FROM_ALL character_set_conv_table_generator.c)
 
-# Custom target that must be manually invoked if character_set_conv_table_generator.c is modified
-add_custom_target(generate_cpl_character_sets
-                  COMMAND $<TARGET_FILE:character_set_conv_table_generator>
-                  WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
-                  DEPENDS character_set_conv_table_generator)
+      if (NOT Iconv_IS_BUILT_IN)
+        if (APPLE)
+          # linkage against OSX's system iconv is going to require LIBICONV_PLUG to be defined, but other self-built
+          # situations like CondaForge or Homebrew might not need this
+          if (HAVE_ICONV_CLOSE)
+            target_compile_definitions(character_set_conv_table_generator PRIVATE -DLIBICONV_PLUG)
+          endif (HAVE_ICONV_CLOSE)
+        endif (APPLE)
+        target_link_libraries(character_set_conv_table_generator PRIVATE Iconv::Iconv)
+      endif ()
+
+    # Custom target that must be manually invoked if character_set_conv_table_generator.c is modified
+    add_custom_target(generate_cpl_character_sets
+                      COMMAND $<TARGET_FILE:character_set_conv_table_generator>
+                      WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+                      DEPENDS character_set_conv_table_generator)
+endif()

